f/connectors/alerts/alerts_gcs.py

```diff
@@ -124,221 +131,251 @@ def _get_tif_rel_filepath(blob_name, local_file_path, territory_id):
+        An additional file to sync from the GCS bucket.
 
     Returns
     -------
-    set of str
-        the (local) file names that were written.
+    tuple
+        A tuple containing a set of str with the local file names that were written
+        and the alerts metadata content.
 
-    FIXME: Ensure dst_path exists before downloading files.
     FIXME: sync, don't brute-force re-download files that already exist - see #6
     """
+
     bucket = storage_client.bucket(bucket_name)
 
-    # List all files in the GCS bucket
+    # List all files in the GCS bucket in territory_id directory
     prefix = f"{territory_id}/"
     files_to_download = set(blob.name for blob in bucket.list_blobs(prefix=prefix))
 
     assert (
         len(files_to_download) > 0
     ), f"No files found to download in bucket '{bucket_name}' with that prefix."
 
-    # Add metadata csv file at the bucket level
-    # TODO: once we have SAs per community and folder, they would need object level read access for this file
-    files_to_download.add(alerts_metadata_filename)
-
     logger.info(
         f"Found {len(files_to_download)} files to download from bucket '{bucket_name}'."
     )
 
-    # Download new or updated files from the GCS bucket to the local directory
+    Path(destination_path).mkdir(parents=True, exist_ok=True)
+
+    downloaded_files = set()
+
+    # Download files from the GCS bucket to the local directory
     for blob_name in files_to_download:
         blob = bucket.blob(blob_name)
-        local_file_path = os.path.join(dst_path, blob_name)
-        filename = local_file_path.split("/")[-1]
+        local_file_path = Path(destination_path) / blob_name
+        filename = local_file_path.name
 
         logger.info(f"Downloading file: {filename}")
 
-        if filename.lower().endswith(".tif") or filename.lower().endswith(".tiff"):
-            rel_filepath = os.path.join(
-                dst_path,
-                _get_tif_rel_filepath(blob_name, local_file_path, territory_id),
-            )
-        # if non-tif such as csv or geojson
-        else:
-            rel_filepath = os.path.dirname(local_file_path)
+        # Generate relative file path for all files
+        rel_filepath = Path(destination_path) / _get_rel_filepath(
+            str(local_file_path), territory_id
+        )
+
+        if not rel_filepath.exists():
+            rel_filepath.mkdir(parents=True, exist_ok=True)
+
+        blob.download_to_filename(rel_filepath / filename)
 
-        if not os.path.exists(rel_filepath):
-            os.makedirs(rel_filepath)
+        downloaded_files.add(str(rel_filepath / filename))
 
-        blob.download_to_filename(os.path.join(rel_filepath, filename))
+    # Create lists of GeoJSON and TIFF files
+    geojson_files = [f for f in downloaded_files if f.endswith(".geojson")]
+    tiff_files = [f for f in downloaded_files if f.endswith((".tif", ".tiff"))]
```

Comment:
Do you find it strange that this function might download a file and then not return its name?  (This would happen if GCS contains a file that's not geojson or tiff).

Tempted to suggest that this check go BEFORE you download, and not download files that don't match?

Reply:
Also the old code's extension comparison was case-insensitive, but your new code's is case-sensitive.